unit-tests/algo/convertions/test-y411.cpp

```diff
@@ -0,0 +1,142 @@
+    int32_t e = yuv.data[2] - 128;
+
+    int32_t t;
+#define clamp( x ) ( ( t = ( x ) ) > 255 ? 255 : t < 0 ? 0 : t )
+    rgb->data[0] = clamp((298 * c + 409 * e + 128) >> 8);
+    rgb->data[1] = clamp((298 * c - 100 * d - 208 * e + 128) >> 8);
+    rgb->data[2] = clamp((298 * c + 516 * d + 128) >> 8);
+#undef clamp
+}
+
+void convert_yuv_to_rgb(byte3 *rgb, const byte3 * yuv)
+{
+    for (auto i = 0; i < W*H; i++)
+    {
+        convert_single_yuv_to_rgb(yuv[i], &rgb[i]);
+    }
+}
+
+// to simulate _mm_mulhi_epi16
+int16_t mult_hi(int16_t x, int16_t y)
+{
+    auto x_i = (int)x;
+    auto y_i = (int)y;
+
+    auto tmp = x_i * y_i;
+
+    auto res = tmp >> 16;
+    return tmp >> 16;
+}
+
+void convert_single_yuv_to_rgb_sse(const byte3 yuv, byte3 *rgb)
+{
+    int16_t c = (yuv.data[0] - 16)<<4;
+    int16_t d = (yuv.data[1] - 128)<<4;
+    int16_t e = (yuv.data[2] - 128)<<4;
+
+    int16_t t;
+#define clamp( x ) ( ( t = ( x ) ) > 255 ? 255 : t < 0 ? 0 : t )
+    rgb->data[0] = clamp(mult_hi((298 << 4) , c) + mult_hi((409<<4) , e) );
+    rgb->data[1] = clamp(mult_hi((298<<4) , c) - mult_hi((100<<4), d) - mult_hi((208<<4), e) );
+    rgb->data[2] = clamp(mult_hi((298<<4) , c) + mult_hi((516<<4) , d));
+#undef clamp
+}
+
+void convert_yuv_to_rgb_sse(byte3 *rgb, const byte3 * yuv)
+{
+    for (auto i = 0; i < W*H; i++)
+    {
+        convert_single_yuv_to_rgb_sse(yuv[i], &rgb[i]);
+    }
+}
+
+byte3 const yuv[W*H] =
+{
+    {10, 0, 30}, {20, 0, 30},    {70, 60, 90},  {80, 60, 90},    {130, 120, 150}, {140, 120, 150},  {190, 180, 210}, {200, 180, 210},   {250, 240, 270}, {260, 240, 270},   {310, 300, 330}, {320, 300, 330},   {370, 360, 390}, {380, 360, 390},   {430, 420, 450}, {440, 420, 450},
+    {40, 0, 30}, {50, 0, 30},    {100, 60, 90}, {110, 60, 90},   {160, 120, 150}, {170, 120, 150},  {220, 180, 210}, {230, 180, 210},   {280, 240, 270}, {290, 240, 270},   {340, 300, 330}, {350, 300, 330},   {400, 360, 390}, {410, 360, 390},   {460, 420, 450}, {470, 420, 450}
+};
+
+#if defined __SSSE3__
+TEST_CASE("unpack_y411_sse")
+{
+    byte3 dest[DEST_BYTES];
+    byte source[SOURCE_BYTES_Y411];
+
+    for (auto i = 0; i < SOURCE_BYTES_Y411; i++)
+    {
+        source[i] = i*10;
+    }
+
+    librealsense::unpack_y411_sse(&((byte*)dest)[0], source, W, H, SOURCE_BYTES_Y411);
+    byte3 rgb[W*H] = { 0 };
+
+    convert_yuv_to_rgb_sse(&rgb[0], &yuv[0]);
+    for (auto i = 0; i < W*H; i++)
+    {
+        CAPTURE(i);
+        CHECK(dest[i] == rgb[i]);
+    }
+}
+
+#endif
+TEST_CASE("unpack_y411_native")
+{
+    byte3 dest[DEST_BYTES];
+    byte source[SOURCE_BYTES_Y411];
+
+    for (auto i = 0; i < SOURCE_BYTES_Y411; i++)
+    {
+        source[i] = i*10;
+    }
+
+    librealsense::unpack_y411_native(&((byte*)dest)[0], source, W, H, SOURCE_BYTES_Y411);
+
+    byte3 rgb[W*H] = { 0 };
+
+    convert_yuv_to_rgb(&rgb[0], &yuv[0]);
+
+    for (auto i = 0; i < W*H; i++)
+    {
+        CAPTURE(i);
+        CHECK(dest[i] == rgb[i]);
```

Comment:
This needs some explanation... maybe I'm tired, and I know you're comparing something to something, but I have no idea what the relationship between `source` and `yuv` is...